lib/io/hardware_can.cpp

```diff
@@ -0,0 +1,132 @@
+#include "hardware_can.hpp"
+
+#include <stdio.h>
+#include <unistd.h>
+
+#include <net/if.h>
+#include <sys/ioctl.h>
+#include <sys/socket.h>
+
+namespace hyped::io {
+
+std::optional<std::shared_ptr<HardwareCan>> HardwareCan::create(
+  core::ILogger &logger, const std::string &can_network_interface)
+{
+  std::int16_t socket_id = socket(PF_CAN, SOCK_RAW, CAN_RAW);
```

Comment:
`const`

also this returns `int`

https://www.man7.org/linux/man-pages/man2/socket.2.html